clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureBaseClient.groovy

```diff
@@ -18,92 +18,39 @@ package com.netflix.spinnaker.clouddriver.azure.client
 
 import com.fasterxml.jackson.databind.DeserializationFeature
 import com.fasterxml.jackson.databind.ObjectMapper
-import com.microsoft.windowsazure.Configuration
-import com.microsoft.windowsazure.exception.ServiceException
-import com.microsoft.windowsazure.management.configuration.ManagementConfiguration
-import com.netflix.spinnaker.clouddriver.azure.common.AzureUtilities
-import com.netflix.spinnaker.clouddriver.azure.security.AzureCredentials
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.microsoft.azure.credentials.ApplicationTokenCredentials
+import com.microsoft.azure.credentials.AzureEnvironment
 import groovy.transform.CompileStatic
 import groovy.util.logging.Slf4j
-import org.apache.commons.io.IOUtils
-import org.apache.http.HttpHost
-import org.apache.http.HttpResponse
-import org.apache.http.HttpStatus
-import org.apache.http.client.methods.HttpGet
-import org.apache.http.impl.client.HttpClientBuilder
+
 
 @Slf4j
 @CompileStatic
 public abstract class AzureBaseClient {
   final String subscriptionId
 
-  static ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+  static ObjectMapper mapper
 
+  /**
+   * Constructor
+   * @param subscriptionId - the Azure subscription to use
+   */
   protected AzureBaseClient(String subscriptionId) {
     this.subscriptionId = subscriptionId
+    mapper = new ObjectMapper().configure(SerializationFeature.INDENT_OUTPUT, true)
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
   }
 
-  protected Configuration buildConfiguration(AzureCredentials creds) throws Exception {
-    try {
-      String baseUri = creds.MANAGEMENT_URL
-      return ManagementConfiguration.configure(
-        null,
-        new Configuration(),
-        new URI(baseUri),
-        this.subscriptionId,
-        creds.getAccessToken())
-    } catch (Exception e) {
-      throw new RuntimeException("Failed to create Azure Resource Management Service", e)
-    }
+  /**
+   * Create the application token credentials object to use for calls to Azure
+   * @param clientId
+   * @param tenantId
+   * @param secret
+     * @return
+     */
+  public static ApplicationTokenCredentials getTokenCredentials(String clientId, String tenantId, String secret) {
```

Comment:
Can drop the `public` here (http://groovy-lang.org/style-guide.html#_public_by_default).